pkg/controller/storageprofile-controller.go

```diff
@@ -86,10 +86,13 @@ func (r *StorageProfileReconciler) reconcileStorageProfile(sc *storagev1.Storage
 
 	storageProfile.Status.StorageClass = &sc.Name
 	storageProfile.Status.Provisioner = &sc.Provisioner
-	snapClass, err := cc.GetSnapshotClassForSmartClone("", &sc.Name, r.log, r.client)
+	snapClass, err := cc.GetSnapshotClassForSmartClone("", &sc.Name, storageProfile.Spec.SnapshotClass, r.log, r.client)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+	if snapClass != "" {
+		storageProfile.Status.SnapshotClass = &snapClass
```

Comment:
Just making sure, status will always be populated with the vsc that is going to be used?
So an Admin can only change the spec unless the one that got picked up is wrong?